MSAL/test/integration/native_auth/end_to_end/MSALNativeAuthEndToEndBaseTestCase.swift

```diff
@@ -26,54 +26,65 @@ import XCTest
-    }
-
-    func mockResponse(_ response: MockAPIResponse, endpoint: MockAPIEndpoint) async throws {
-        try await mockAPIHandler.addResponse(
-            endpoint: endpoint,
-            correlationId: correlationId,
-            responses: [response]
-        )
+    class Constants {
+        static let nativeAuthKey = "native_auth"
+        static let clientIdEmailPasswordKey = "email_password_client_id"
+        static let clientIdEmailCodeKey = "email_code_client_id"
+        static let clientIdEmailPasswordAttributesKey = "email_password_attributes_client_id"
+        static let clientIdEmailCodeAttributesKey = "email_code_attributes_client_id"
+        static let tenantSubdomainKey = "tenant_subdomain"
+        static let signInEmailPasswordUsernameKey = "sign_in_email_password_username"
+        static let signInEmailCodeUsernameKey = "sign_in_email_code_username"
+        static let resetPasswordUsernameKey = "reset_password_username"
     }
+    
+    let correlationId = UUID()
+    let defaultTimeout: TimeInterval = 20
 
-    override func tearDown() {
-        try? mockAPIHandler.clearQueues(correlationId: correlationId)
-    }
+    private var confFileContent: [String: String]? = nil
+    private let codeRetriever = MSALNativeAuthEmailCodeRetriever()
 
     override func setUpWithError() throws {
         try super.setUpWithError()
-        let useMockAPIBooleanString = ProcessInfo.processInfo.environment["useMockAPI"] ?? "false"
-        usingMockAPI = Bool(useMockAPIBooleanString) ?? false
         
-        // mock API URL needs to contains a tenant
-        guard let authorityURL = URL(string: Configuration.authorityURLString + (usingMockAPI ? "/testTenant" : "")), let authority = try? MSALCIAMAuthority(url: authorityURL) else {
-            XCTFail("AuthorityURL not set or invalid")
+        guard let confURL = Bundle(for: Self.self).url(forResource: "conf", withExtension: "json"), let configurationData = try? Data(contentsOf: confURL) else {
+            XCTFail("conf.json file not found")
             return
         }
-
-        sut = try MSALNativeAuthPublicClientApplication(
-            configuration: MSALPublicClientApplicationConfig(
-                clientId: Configuration.clientId,
-                redirectUri: nil,
-                authority: authority
-            ),
-            challengeTypes: [.OOB, .password]
-        )
-
-        if usingMockAPI {
-            print("🤖🤖🤖 Using mock API: \(Configuration.authorityURLString)")
-        } else {
-            print("👩‍💻👩‍💻👩‍💻 Using test tenant: \(Configuration.authorityURLString)")
+        let confFile = try? JSONSerialization.jsonObject(with: configurationData, options: []) as? [String: Any]
+        confFileContent = confFile?[Constants.nativeAuthKey] as? [String: String]
+    }
+    
+    func initialisePublicClientApplication(
+        useEmailPasswordClientId: Bool = true,
```

Comment:
I wouldn't set a default value to `useEmailPasswordClientId: Bool = true,`, as we have an entirely set of tests (the emailAndCode ones) for which this is false.

In fact, I would use an enum rather an an boolean. I think this is what @spetrescu84 is saying in [his comment](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2217/files#r1654979578).

Reply:
Can you please suggest how this will look in the code? If we have to define all scenarios as enum cases, the enum might become quite large. For instance, we would need to include 4 use cases for this particular scenario. However, it's possible I might have misunderstood your suggestion.

Reply:
I think we can just differentiate between email otp and email with password for now.

```
enum Userflow {
    case emailOtp
    case emailPassword
}
```

I named it `UserFlow` to match what is in the portal, but feel free to use another name.

Reply:
Thanks for the suggestion. Let's wait for Silviu's input because he suggested something similar in a previous comment.